ci/docker/ml.build.Dockerfile

```diff
@@ -21,10 +22,16 @@ RUN <<EOF
 
 set -euo pipefail
 
-DOC_TESTING=1 TRAIN_TESTING=1 TUNE_TESTING=1 DATA_PROCESSING_TESTING=1 INSTALL_HDFS=1 \
-  ./ci/env/install-dependencies.sh
+if [[ "${PYTHON-}" == "3.12" ]]; then
```

Comment:
maybe separate this file in another PR? I can quickly approve this change.

Reply:
let's do it